### PR TITLE
elizCurtnazar/tema8

### PR DESCRIPTION
**Pull request by curtnazarEliz**

### CODE REVIEW

File: src/test/java/elizCurtnazar/tests/tema8/pageObject/DisableDynamicButtonsPage.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    Wait<WebDriver> fluentWait;
+
+    public DisableDynamicButtonsPage(ChromeDriver driver) {
+        this.driver = driver;
+        //using WebDriverWait
+        wait = new WebDriverWait(driver, Duration.ofSeconds(9));
+        //using FluentWait
+        fluentWait = new FluentWait<WebDriver>(driver)
+                .withTimeout(Duration.ofSeconds(1))
+                .pollingEvery(Duration.ofSeconds(8));
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/pageObject/DisableDynamicButtonsPage.java:31]: It doesn't make any sense to try after 8 seconds when maximum timeout is 1 second. Please use a smaller unit than seconds for pollingEvery part as you used in other page classes. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/pageObject/DisableDynamicButtonsPage.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    WebElement buttonOne;
+
+    @FindBy(id = "button02")
+    WebElement buttonTwo;
+
+    @FindBy(id = "button03")
+    WebElement buttonThree;
+
+    public void clickButtons() {
+        Assert.assertEquals(getMessage(), "Click Buttons In Order");
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/pageObject/DisableDynamicButtonsPage.java:51]: Please split this method into different methods, one per each buttons and move assertions from page object to test level. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/pageObject/JavaScriptLandingPage.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    Wait<WebDriver> fluentWait;
+
+    public JavaScriptLandingPage(ChromeDriver driver) {
+        this.driver = driver;
+        //using WebDriverWait
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        //using FluentWait
+        fluentWait = new FluentWait<WebDriver>(driver)
+                .withTimeout(Duration.ofSeconds(1))
+                .pollingEvery(Duration.ofSeconds(8));
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/pageObject/JavaScriptLandingPage.java:26]: It doesn't make any sense to try after 8 seconds when maximum timeout is 1 second. Please use a smaller unit than seconds for pollingEvery part as you used in other page classes. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/pageObject/JavaScriptRedirectsPage.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    Wait<WebDriver> fluentWait;
+
+    public JavaScriptRedirectsPage(ChromeDriver driver) {
+        this.driver = driver;
+        //using WebDriverWait
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        //using FluentWait
+        fluentWait = new FluentWait<WebDriver>(driver)
+                .withTimeout(Duration.ofSeconds(1))
+                .pollingEvery(Duration.ofSeconds(8));
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/pageObject/JavaScriptRedirectsPage.java:28]: It doesn't make any sense to try after 8 seconds when maximum timeout is 1 second. Please use a smaller unit than seconds for pollingEvery part as you used in other page classes. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/pageObject/RefreshPage.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    WebDriverWait wait;
+    Wait<WebDriver> fluentWait;
+    public RefreshPage(ChromeDriver driver) {
+        this.driver = driver;
+        //using WebDriverWait
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        //using FluentWait
+        fluentWait = new FluentWait<WebDriver>(driver)
+                .withTimeout(Duration.ofSeconds(1))
+                .pollingEvery(Duration.ofSeconds(8));
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/pageObject/RefreshPage.java:25]: It doesn't make any sense to try after 8 seconds when maximum timeout is 1 second. Please use a smaller unit than seconds for pollingEvery part as you used in other page classes. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/pageObject/SimpleDynamicButtonsPage.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    WebDriverWait wait;
+    Wait<WebDriver> fluentWait;
+    public SimpleDynamicButtonsPage(ChromeDriver driver) {
+        this.driver = driver;
+        //using WebDriverWait
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        //using FluentWait
+        fluentWait = new FluentWait<WebDriver>(driver)
+                .withTimeout(Duration.ofSeconds(1))
+                .pollingEvery(Duration.ofSeconds(8));
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/pageObject/SimpleDynamicButtonsPage.java:26]: It doesn't make any sense to try after 8 seconds when maximum timeout is 1 second. Please use a smaller unit than seconds for pollingEvery part as you used in other page classes. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+public class BaseTest {
+
+    protected ChromeDriver driver;
+
+    @BeforeMethod(groups = "desktop")
+    public void initChromeBrowser() {
+        driver = BrowserManager.createChromeDriver();
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java:19]: Call method for creating chrome driver maximized from BrowserManager class and remove the other unnecessary lines from this method. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+//        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
+        driver.manage().window().maximize();
+        System.out.println("Executed before method");
+
+    }
+    @BeforeMethod(groups = "mobile")
+    public void initChromeBrowserForMobile() {
+        Map<String, String> mobileEmulation = new HashMap<>();
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java:27]: Call method for creating chrome driver for mobile from BrowserManager class and remove the other unnecessary lines from this method. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/teste/BasicAjaxTests.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Test(groups = {"mobile","desktop"})
+    public void codeInItTest2() {
+        driver.get("https://testpages.herokuapp.com/styled/basic-ajax-test.html");
+        BasicAjaxPage basicAjaxPage = new BasicAjaxPage(driver);
+        basicAjaxPage.selectCategory("Web");
+        basicAjaxPage.selectLanguage("Flash");
+        basicAjaxPage.clickOnSubmitButton();
+        BasicAjaxConfirmationPage basicAjaxConfirmationPage = new BasicAjaxConfirmationPage(driver);
+        System.out.println(basicAjaxConfirmationPage.getLanguageType());
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/teste/BasicAjaxTests.java:61]: You're missing an assertion here for this scenario. Please add it. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    @Test(groups = {"mobile","desktop"})
+    public void clickFirstButton() {
+           driver.get("https://testpages.herokuapp.com/styled/javascript-redirect-test.html");
+           WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+           JavaScriptRedirectsPage javaScriptRedirectsPage = new JavaScriptRedirectsPage(driver);
+           javaScriptRedirectsPage.getParagraphText();
+           javaScriptRedirectsPage.clickFirstButton();
+           JavaScriptLandingPage javaScriptLandingPage = new JavaScriptLandingPage(driver);
+           javaScriptLandingPage.getTextofParagraph();
+           javaScriptLandingPage.clickGoBackButton();
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java:21]: You're missing an assertion here for this scenario. Please add it. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+     @Test(groups = {"mobile","desktop"})
+    public void clickSecondButton() {
+         driver.get("https://testpages.herokuapp.com/styled/javascript-redirect-test.html");
+         WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+         JavaScriptRedirectsPage javaScriptRedirectsPage = new JavaScriptRedirectsPage(driver);
+         javaScriptRedirectsPage.getParagraphText();
+         javaScriptRedirectsPage.clickSecondButton();
+         JavaScriptLandingPage javaScriptLandingPage = new JavaScriptLandingPage(driver);
+         javaScriptLandingPage.getTextofParagraph();
+         javaScriptLandingPage.clickGoBackButton();
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java:33]: You're missing an assertion here for this scenario. Please add it. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/teste/RefreshPageTests.java [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+public class RefreshPageTests extends BaseTest {
+
+    @Test(groups = {"mobile","desktop"})
+    public void refreshPage() {
+        driver.get("https://testpages.herokuapp.com/styled/refresh");
+        RefreshPage refreshPage = new RefreshPage(driver);
+        refreshPage.getTimeOfTheRefresh();
+        System.out.println("Am intrat pe pagine de refresh si timpul este " + refreshPage.getTimeOfTheRefresh());
+        driver.navigate().refresh();
+        System.out.println("Am dat un refresh si timpul acum este " + refreshPage.getTimeOfTheRefresh());
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/teste/RefreshPageTests.java:15]: You're missing an assertion here for this scenario. Please add it. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/teste/SimpleDynamicButtonsTests.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
+package elizCurtnazar.tests.tema8.teste;
+
+import elizCurtnazar.tests.tema8.pageObject.SimpleDynamicButtonsPage;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+//import teofilursan.tests.tema8.pageobjects.BasicAjaxConfirmationPage;
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/teste/SimpleDynamicButtonsTests.java:6]: Please remove these commented lines. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]
Don't forget to add .xml testng files for desktop and mobile. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/pageObject/DisableDynamicButtonsPage.java (hunk 1 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema8/pageObject/DisableDynamicButtonsPage.java]: done

File: src/test/java/elizCurtnazar/tests/tema8/pageObject/DisableDynamicButtonsPage.java (hunk 2 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema8/pageObject/DisableDynamicButtonsPage.java]: done

File: src/test/java/elizCurtnazar/tests/tema8/pageObject/JavaScriptLandingPage.java (hunk 3 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema8/pageObject/JavaScriptLandingPage.java]: done

File: src/test/java/elizCurtnazar/tests/tema8/pageObject/RefreshPage.java (hunk 5 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema8/pageObject/RefreshPage.java]: done

File: src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java (hunk 7 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java]: done

File: src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java (hunk 8 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java:28]: aici nu stiu cum sa fac

File: src/test/java/elizCurtnazar/tests/tema8/teste/BasicAjaxTests.java (hunk 9 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema8/teste/BasicAjaxTests.java:61]: done

File: src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java (hunk 10 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java]: done

File: src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java (hunk 11 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java]: done

File: src/test/java/elizCurtnazar/tests/tema8/teste/RefreshPageTests.java (hunk 12 above)
Comment (curtnazarEliz) [reply src/test/java/elizCurtnazar/tests/tema8/teste/RefreshPageTests.java:16]: done

File: src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+public class BaseTest {
+
+    protected ChromeDriver driver;
+
+    @BeforeMethod(groups = "desktop")
+    public void initChromeBrowser() {
+        driver = new ChromeDriver();
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java:19]: Replace all of the lines from this method with ->
driver = BrowserManager.createChromeDriverWithOptions();

File: src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java (hunk 8 above)
Comment (95teo) [reply src/test/java/elizCurtnazar/tests/tema8/teste/BaseTest.java:28]: Replace all of the lines from this method with ->
driver = BrowserManager.createChromeDriverForMobile();

File: src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+
+public class JavaScriptRedirectsTests extends BaseTest {
+
+    @Test(groups = {"mobile", "desktop"})
+    public void clickFirstButton() {
+        driver.get("https://testpages.herokuapp.com/styled/javascript-redirect-test.html");
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java:16]: You can remove this line as you're not using anywhere. Thanks!

File: src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+        javaScriptRedirectsPage.clickFirstButton();
+        JavaScriptLandingPage javaScriptLandingPage = new JavaScriptLandingPage(driver);
+        Assert.assertEquals(javaScriptLandingPage.getTextofParagraph(), "You have been successfully redirected.");
+        javaScriptLandingPage.clickGoBackButton();
+    }
+
+    @Test(groups = {"mobile", "desktop"})
+    public void clickSecondButton() {
+        driver.get("https://testpages.herokuapp.com/styled/javascript-redirect-test.html");
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
```
Comment (95teo) [inline src/test/java/elizCurtnazar/tests/tema8/teste/JavaScriptRedirectsTests.java:28]: You can remove this line as you're not using anywhere. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

**Review by 95teo — APPROVED** [review]

**Review by tincageorgianaalina — APPROVED** [review]

**Pull request merged**